kernel/src/scan/state.rs

```diff
@@ -98,14 +98,17 @@ impl DvInfo {
     }
 }
 
-pub type ScanCallback<T> = fn(
-    context: &mut T,
-    path: &str,
-    size: i64,
-    stats: Option<Stats>,
-    dv_info: DvInfo,
-    partition_values: HashMap<String, String>,
-);
+/// A struct containing all the information needed for a scan file callback
+#[derive(Debug, Clone)]
+pub struct ScanFile {
+    pub path: String,
```

Comment:
I went down the lifetime rabbithole to see if i could keep it as &str and it ended up being really complicated and doubtful it makes a difference perf wise.

Reply:
Yeah, we were copying it anyway before, just one layer lower